lib/index.coffee

```diff
@@ -89,13 +88,12 @@ class EPub
-      allowedV2Tags = ["div", "p", "h1", "h2", "h3", "h4", "h5", "h6", "ul", "ol", "li", "dl", "dt", "dd", "address", "hr", "pre", "blockquote", "center", "ins", "del", "a", "span", "bdo", "br", "em", "strong", "dfn", "code", "samp", "kbd", "bar", "cite", "abbr", "acronym", "q", "sub", "sup", "tt", "i", "b", "big", "small", "u", "s", "strike", "basefont", "font", "object", "param", "img", "table", "caption", "colgroup", "col", "thead", "tfoot", "tbody", "tr", "th", "td", "embed", "applet", "iframe", "img", "map", "noscript", "ns:svg", "object", "script", "table", "tt", "var"]
-#      content.data = require('xhtml-purifier').purify(content.data);
+      allowedXhtml11Tags = ["div", "p", "h1", "h2", "h3", "h4", "h5", "h6", "ul", "ol", "li", "dl", "dt", "dd", "address", "hr", "pre", "blockquote", "center", "ins", "del", "a", "span", "bdo", "br", "em", "strong", "dfn", "code", "samp", "kbd", "bar", "cite", "abbr", "acronym", "q", "sub", "sup", "tt", "i", "b", "big", "small", "u", "s", "strike", "basefont", "font", "object", "param", "img", "table", "caption", "colgroup", "col", "thead", "tfoot", "tbody", "tr", "th", "td", "embed", "applet", "iframe", "img", "map", "noscript", "ns:svg", "object", "script", "table", "tt", "var"]
       content.data = content.data.replace(/\&/g, "&amp;")
       $ = cheerio.load content.data, {xmlMode: true, lowerCaseTags: true, ignoreWhitespace: true, recognizeSelfClosing: true}
       if $("body").length
         $ = cheerio.load $("body").html(), {xmlMode: true, lowerCaseTags: true, ignoreWhitespace: true, recognizeSelfClosing: true}
-      $("*").each (index, elem)->
+      $($("*").get().reverse()).each (elemIndex, elem)->
```

Comment:
We had an issue here, because since the jQuery/cheerio selector works/fetches the DOM elements from top-to-bottom, would pick the top elements first, and swap the contents if it was a invalid tag, which by doing so, the following previous nested elements (further on the array) would be disconnected from the original parent because the parent was updated with new contents.

Reversing the order fixes it because this way the update is moving from bottom-to-top on the DOM structure, and swapping and handling contents properly.

🚀